### DISCUSSION TITLE
Prevent BootstrapEmail from URL encoding URLs?

### DISCUSSION THREAD

**Discussion by ComtelJeremy** [bootstrap-email/bootstrap-email#131] [Help] [answered]
Hello, we use bootstrap e-mail to compile our e-mail templates. They include links that are later injected with variables before being sent...

Unfortunately, we use a handlebars style template, and when bootstrap e-mail is done bootstrapifying the template the {{ and }}  in the <a> tag become %7B%7B and %7D%7D. Then when we go to inject the template, these handlebar tags are missed.

For now, we might just revert  the %7Bs and %7Ds to {{ and }} with a find / replace after bootstrapifying, but it would be nice to be able to disable this behavior.

Any help would be appreciated!

**Comment by stuyam** [comment]
Yeah this is on my list of things I want to improve. I don't have a solution right now but have some ideas. I was thinking of adding some kind of special tag to "preserve content" which can be trickier than it sounds. This is something that nokogiri does automatically to links, so maybe I could just build support to support handlebar like syntax in an href specifically.

This only happens in the href right? This doesn't happen to handlebars elsewhere in the code right?

**Reply by ComtelJeremy** [reply]
Correct... So far it only happens in in <a> tag (href). I haven't checked an img since it hasn't come up yet (but will soon in our new templates). For now I am doing a find replace when I load our templates, but it would be nice to strip that out later.

**Reply by deckster271** [reply]
Hi there - just to add, this also happens in img src's.
I'm having the same problem 

> Yeah this is on my list of things I want to improve. I don't have a solution right now but have some ideas. I was thinking of adding some kind of special tag to "preserve content" which can be trickier than it sounds. This is something that nokogiri does automatically to links, so maybe I could just build support to support handlebar like syntax in an href specifically.
> 
> This only happens in the href right? This doesn't happen to handlebars elsewhere in the code right?

**Reply by deckster271** [reply]
Oh, ignore my reply - I can see you've also fixed it for img src's in v1.0.2

Thanks 👍

**Comment by stuyam** [comment, marked as answer]
@ComtelJeremy this is fixed in [v1.0.2](https://github.com/bootstrap-email/bootstrap-email/releases/tag/1.0.2). Give it a try, deploying to the online editor now as well.

**Reply by ComtelJeremy** [reply]
Awesome... we'll try it in the next week and let you know.

**Reply by PneutronixJeremy** [reply]
Hello again @stuyam !
This was fixed, but came up again.
It seems now when we have two handlebars references in a <a> tag, it url encodes the seconds one. Do we need to adjust the regex to allow multiple matches?

Here is an example of the issue:

``` <a href="https://{{Domain}}/{{shortGuid MessageGuid}}"> ```
Becomes:
``` <a href="[https://{{Domain}}/%7B%7BshortGuid%20MessageGuid%7D%7D"> ```

Any suggestions? Sorry we didn't catch this earlier.
